sys/ztimer/auto_init.c

```diff
@@ -111,6 +111,11 @@ void ztimer_init(void)
                              CONFIG_ZTIMER_USEC_BASE_FREQ,
                              WIDTH_TO_MAXVAL(CONFIG_ZTIMER_USEC_WIDTH));
 #  endif
+#  ifdef MODULE_PM_LAYERED
+    LOG_DEBUG("ztimer_init(): ZTIMER_USEC setting required_pm_mode to %i\n",
+              CONFIG_ZTIMER_USEC_REQUIRED_PM_MODE);
+    ZTIMER_USEC_BASE->required_pm_mode = CONFIG_ZTIMER_USEC_REQUIRED_PM_MODE;
+#  endif
```

Comment:
This section can cause errors during compilation if `CONFIG_ZTIMER_USEC_REQUIRED_PM_MODE` is configured with the wrongly typed value:

```
In file included from /home/jue/Projects/RIOT/sys/ztimer/auto_init.c:48:
/home/jue/Projects/RIOT/sys/ztimer/auto_init.c: In function 'ztimer_init':
/home/jue/Projects/RIOT/sys/ztimer/auto_init.c:115:15: error: format '%i' expects argument of type 'int', but argument 2 has type 'long unsigned int' [-Werror=format=]
  115 |     LOG_DEBUG("ztimer_init(): ZTIMER_USEC setting required_pm_mode to %i\n",
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This happens, if use the defines made by the cpu vendor.

I'd say the easy solution is to set `ZTIMER_USEC_BASE->required_pm_mode` and then print that value. If I'm familiar enough with GCC and clang this should fix casting issues due to implicit casting.

```suggestion
#  ifdef MODULE_PM_LAYERED
    ZTIMER_USEC_BASE->required_pm_mode = CONFIG_ZTIMER_USEC_REQUIRED_PM_MODE;
    LOG_DEBUG("ztimer_init(): ZTIMER_USEC setting required_pm_mode to "PRIu8"\n",
              ZTIMER_USEC_BASE->required_pm_mode);
#  endif
```

The same goes with the respective MSEC section.

What do you think?
Maybe @kaspar030 has an opinion on this, too?

Reply:
If you apply this, please use the past tense in the debug message. (setting -> set)